test/cpp/api/modules.cpp

```diff
@@ -605,6 +605,61 @@ TEST_F(ModulesTest, MaxUnpool2d) {
   ASSERT_EQ(y.sizes(), torch::IntArrayRef({2, 1, 5, 5}));
 }
 
+TEST_F(ModulesTest, MaxUnpool3d) {
+  auto indices = torch::tensor({{{{{26}}}}}, torch::kLong);
+  auto x = torch::tensor({{{{{26}}}}}, torch::requires_grad());
+  auto model = MaxUnpool3d{3};
+  auto y = model->forward(x, indices);
+
+  ASSERT_EQ(y.dim(), 5);
+  ASSERT_TRUE(torch::allclose(y, torch::tensor(
+   {{{{{ 0,  0,  0},
+       { 0,  0,  0},
+       { 0,  0,  0}},
+      {{ 0,  0,  0},
+       { 0,  0,  0},
+       { 0,  0,  0}},
+      {{ 0,  0,  0},
+       { 0,  0,  0},
+       { 0,  0, 26}}}}}, torch::kFloat)));
+  ASSERT_EQ(y.sizes(), torch::IntArrayRef({1, 1, 3, 3, 3}));
+}
+
+TEST_F(ModulesTest, MaxUnpool3dOutputSize) {
+  auto indices = torch::tensor(
+    {{{{{21, 23},
+        {29, 31}},
+       {{53, 55},
+        {61, 63}}}}}, torch::kLong);
+    auto x = torch::tensor(
+    {{{{{21, 23},
+        {29, 31}},
+       {{53, 55},
+        {61, 63}}}}}, torch::requires_grad());
+  auto model = MaxUnpool3d{MaxUnpool3dOptions(3).stride(2).padding(1)};
+  auto y = model->forward(x, indices, torch::IntArrayRef({1, 1, 4, 4, 4}));
+
+  ASSERT_EQ(y.dim(), 5);
+  ASSERT_TRUE(torch::allclose(y, torch::tensor(
+   {{{{{ 0,  0,  0,  0},
+       { 0,  0,  0,  0},
+       { 0,  0,  0,  0},
+       { 0,  0,  0,  0}},
+      {{ 0,  0,  0,  0},
+       { 0, 21,  0, 23},
+       { 0,  0,  0,  0},
+       { 0, 29,  0, 31}},
+      {{ 0,  0,  0,  0},
+       { 0,  0,  0,  0},
+       { 0,  0,  0,  0},
+       { 0,  0,  0,  0}},
+      {{ 0,  0,  0,  0},
+       { 0, 53,  0, 55},
+       { 0,  0,  0,  0},
+       { 0, 61,  0, 63}}}}}, torch::kFloat)));
+  ASSERT_EQ(y.sizes(), torch::IntArrayRef({1, 1, 4, 4, 4}));
+}
```

Comment:
It would be great to add a test case for MaxPool3d -> MaxUnpool3d, similar to the 1d and 2d cases.

Reply:
done